File: Dumplings/src/main/java/net/sinclairstudios/util/TextViewUpdater.java
package net.sinclairstudios.util;

import android.widget.TextView;

import java.text.DecimalFormat;

public class TextViewUpdater {
    private final TextView textView;
    private final String fieldValue;

    public TextViewUpdater(TextView textView) {
        this.textView = textView;
        this.fieldValue = textView.getText().toString();
    }

    public void update(int value) {
        textView.setText(fieldValue.replace("{}", DecimalFormat.getInstance().format(value)));
    }
}

File: app/src/main/java/net/sinclairstudios/dumplings/util/TextViewUpdater.kt
package net.sinclairstudios.dumplings.util

import android.widget.TextView
import java.text.DecimalFormat

class TextViewUpdater(private val textView: TextView) {
    private val fieldValue = textView.text.toString()

    fun update(value: Int) {
        textView.text = fieldValue.replace("{}", DecimalFormat.getInstance().format(value.toLong()))
    }
}